alpine-consul/root/etc/services.d/consul-template/run

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/with-contenv sh
```

Comment:
> In order to stop consul-template running indefinitely

It is meant to run indefinitely as it monitors changes and updates files accordingly. That is why there's s6 command to restart, see [examples/user-consul-nginx-nodejs/root/etc/consul-template/conf.d/nginx/nginx.conf](https://github.com/matthewvalimaki/docker-alpine/blob/add-consul-template/examples/user-consul-nginx-nodejs/root/etc/consul-template/conf.d/nginx/nginx.conf) and `command = "s6-svc -h /var/run/s6/services/nginx"`.

I will add the `down` file and init script as that is exactly what is needed. Thanks for pointing this feature out!

> Also, perhaps the looping logic could be tidied up by changing this from a shell script to an execline script and utilising s6-svwait

I did try s6-svwait but unfortunately what happens is that while `consul agent` is up it is not usable immediately causing `consul-template` to throw up an error as it tries to query for data for templates. We could try [notification-fd](http://skarnet.org/software/s6/notifywhenup.html) however I am not really sure how to implement this. If you have an example or know how to please let me know.